source/tests/unit/library_set_runtime_environment_tests.cpp

```diff
@@ -20,15 +21,15 @@ TEST(LibrarySetRuntimeEnvironmentTests, Library_SetRuntimeEnvironmentFound_Calls
   EXPECT_CALL(*p_shared_library, set_library_name(_));
   EXPECT_CALL(*p_shared_library, load());
   EXPECT_CALL(*p_shared_library, is_loaded())
-    .WillOnce(Return(true)); // the constructor will bail out early if false
+      .WillOnce(Return(true));  // the constructor will bail out early if false
 
   const char* func_name = "niSwitch_SetRuntimeEnvironment";
 
   EXPECT_CALL(*p_shared_library, get_function_pointer(_))
-    .WillRepeatedly(Return(nullptr));
+      .WillRepeatedly(Return(nullptr));
   // We must return an address for SetRuntimeEnvironment or we won't call it.
-  EXPECT_CALL(*p_shared_library, get_function_pointer(func_name))
-    .WillOnce(Return(const_cast<const void*>(reinterpret_cast<void*>(&FunctionToPointAt))));
+  EXPECT_CALL(*p_shared_library, get_function_pointer(::testing::StrEq(func_name)))
```

Comment:
Note: This was the fix. Without it, locally and on the generic debian 12 container where someone else was running it, it wouldn't match here and we'd just get a nullptr for the "niSwitch_SetRuntimeEnvironment".